spec-tests/src/test/java/org/candlepin/spec/HypervisorCheckInSpecTest.java

```diff
@@ -0,0 +1,882 @@
+                guestConsumer.getUuid()).size());
+
+            // Updating to a new host should remove host specific entitlements
+            // because MySql
+            sleep(2000);
+            hypervisorCheckin(owner, userClient, "tester",
+                hypervisorId2, List.of(uuid1), null, reporterId, true);
+            // The guests host limited entitlement should be gone
+            assertEquals(0, guestConsumerClient.consumers().listEntitlements(guestConsumer.getUuid(),
+                null, true, null, null, null, null, null).size());
+        }
+    }
+
+    private HypervisorUpdateResultDTO hypervisorCheckin(OwnerDTO owner, ApiClient consumerClient,
+        String hypervisorName, String hypervisorId, List<String> guestIds, Map<String, String> facts,
+        String reporterId, boolean createMissing) throws ApiException, IOException {
+        JsonNode hostGuestMapping = getAsyncHostGuestMapping(hypervisorName, hypervisorId, guestIds, facts);
+        AsyncJobStatusDTO job = consumerClient.hypervisors().hypervisorUpdateAsync(
+            owner.getKey(), createMissing, reporterId, hostGuestMapping.toString());
+        HypervisorUpdateResultDTO resultData = null;
+        if (job != null) {
+            AsyncJobStatusDTO status = client.jobs().waitForJob(job.getId());
+            assertEquals("FINISHED", status.getState());
+            resultData = getResultData(status);
+        }
+        return resultData;
+    }
+
+    private void asyncUpdateGuestIdsTest(ApiClient client, String reporter) throws Exception {
+        List<String> updatedGuestIds = List.of(guest2VirtUuid, "Guest3");
+        JsonNode mapping = getAsyncHostGuestMapping(
+            expectedHostName, expectedHostHypervisorId, updatedGuestIds, null);
+        AsyncJobStatusDTO job = client.hypervisors().hypervisorUpdateAsync(
+            owner.getKey(), true, reporter, mapping.toString());
+        HypervisorUpdateResultDTO resultData = null;
+        if (job != null) {
+            AsyncJobStatusDTO status = client.jobs().waitForJob(job.getId());
+            assertEquals("FINISHED", status.getState());
+            resultData = getResultData(status);
+        }
+        confirmResultDataCounts(resultData, 0, 1, 0, 0);
+        ConsumerDTO consumer = consumerApi.getConsumer(hostUuid);
+        checkHypervisorConsumer(consumer, expectedHostName, updatedGuestIds, reporter);
+    }
+
+    /**
+     * The HypervisorUpdateResultDTO fromJson method will not parse the raw string from the
+     *   AsyncJobStatusDTO
+     *
+     * @param status
+     * @return HypervisorUpdateResultDTO
+     * @throws IOException
+     */
+    private HypervisorUpdateResultDTO getResultData(AsyncJobStatusDTO status) throws IOException {
+        Map resultData = (Map) status.getResultData();
+        Gson gson = new GsonBuilder().create();
+        String json = gson.toJson(resultData);
+        return HypervisorUpdateResultDTO.fromJson(json);
+    }
```

Comment:
This could be reduced to a single line using the existing Jaskson ObjectMapper that is defined ApiClient. I don't think this is a blocker though, both ways work!

`ApiClient.MAPPER.convertValue(status.getResultData(), HypervisorUpdateResultDTO.class)`